integreat_cms/cms/utils/linkcheck_utils.py

```diff
@@ -309,3 +291,55 @@ def fix_content_link_encoding(content: str) -> str:
     :return: The fixed content
     """
     return re.sub(r"(?<=[\"'])(https?://.+?)(?=[\"'])", fix_domain_encoding, content)
+
+
+def log_replacement_is_starting(
+    search: str,
+    replace: str,
+    region: Region | None = None,
+    user: User | None = None,
+) -> None:
+    """
+    function to log the current link replacement
+
+    :param search: The (partial) URL to search
+    :param replace: The (partial) URL to replace
+    :param region: Optionally limit the replacement to one region (``None`` means a global replacement)
+    :param user: The creator of the replaced translations
+    """
+    region_msg = f' of "{region!r}"' if region else ""
+    user_msg = f' by "{user!r}"' if user else ""
+    logger.info(
+        "Replacing %r with %r in content links%s%s",
+        search,
+        replace,
+        region_msg,
+        user_msg,
+    )
+
+
+def find_links_and_translations(
+    search: str, region: Region | None = None, link_types: list[str] | None = None
+) -> tuple[list[Link], QuerySet]:
+    """
+    Collects link objects which match the selected link types and contain string given as search in the url,
+    and find translations which contain those links
+
+    :param search: The (partial) URL to search
+    :param region: Optionally limit the replacement to one region (``None`` means a global replacement)
+    :param link_types: Which kind of links should be replaced
+
+    :return: A tuple of links and translations
+    """
+    # This function is used in replace_links, which is used in the management command, where region can be None, too.
```

Comment:
Nit: As this to me is concerning the entire method and not just a single line, I would add it in line 327 between description and types. But this is just my personal preference.